api/models/fn_test.go

```diff
@@ -0,0 +1,109 @@
+package models
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/leanovate/gopter"
+	"github.com/leanovate/gopter/gen"
+	"github.com/leanovate/gopter/prop"
+)
+
+func fnReflectType() reflect.Type {
+	fn := Fn{}
+	return reflect.TypeOf(fn)
+}
+
+func resourceConfigGenerator(t *testing.T) gopter.Gen {
+	fieldGens := make(map[string]gopter.Gen)
+
+	fieldGens["Memory"] = gen.UInt64()
+	fieldGens["Timeout"] = gen.Int32()
+	fieldGens["IdleTimeout"] = gen.Int32()
+
+	resourceConfig := ResourceConfig{}
+	resourceConfigFieldCount := reflect.TypeOf(resourceConfig).NumField()
+
+	if resourceConfigFieldCount != len(fieldGens) {
+		t.Fatalf("Fn struct field count, %d, does not match fn generator field count, %d", resourceConfigFieldCount, len(fieldGens))
+	}
+
+	return gen.Struct(reflect.TypeOf(resourceConfig), fieldGens)
+}
+
+func fnFieldGenerators(t *testing.T) map[string]gopter.Gen {
+	fieldGens := make(map[string]gopter.Gen)
+
+	fieldGens["ID"] = gen.AlphaString()
+	fieldGens["Name"] = gen.AlphaString()
+	fieldGens["AppID"] = gen.AlphaString()
+	fieldGens["Image"] = gen.AlphaString()
+	fieldGens["Config"] = configGenerator()
+	fieldGens["ResourceConfig"] = resourceConfigGenerator(t)
+	fieldGens["Annotations"] = annotationGenerator()
+	fieldGens["CreatedAt"] = datetimeGenerator()
+	fieldGens["UpdatedAt"] = datetimeGenerator()
+	fieldGens["Format"] = gen.AlphaString()
```

Comment:
sorry this has been on the backburner, I'll get better at my flagged emails again one day (send help). maintenance of the fields here concerns me, but the expectation is that if a field gets added or removed then these tests will fail and someone will have to go in and update these tests, right? I'd suppose that's a good thing overall, shouldn't be able to just go add a field willy nilly without having to add tests yea? Roundabout I'm asking whether it's possible to do this without the generators, almost fuzz style (but with working ones), and get the same effect but without the code maintenance overhead here.

Reply:
Yeah, it's kind of the point to highlight that new fields need a bit more than just adding to the struct.

I think you could do something without the generators, but it wouldn't explore as much as this version does. The payoff is slight here, but I'd like to add more of these style tests in the future.